### PR TITLE
Fixed User Location Toggle and Path Focus on Map

### PR DESCRIPTION
**Pull request by cryogon**
The Toggle that make the map zoom on user's current location is now working properly and when the path is selected the map is zooming on correct location instead of just zooming wrong when distance between map's current center and path location is too far

### CODE REVIEW

File: src/components/MapComponent.vue [hunk 1]
```diff
@@ -187,23 +196,37 @@ export default {
       }
     );
   },
+  mounted() {
+    this.emitter.on("pathSelected", (path) => {
```
Comment (blahmonkey) [inline src/components/MapComponent.vue:200]: @JatinThakur14 You can use `path => {` without `(path)`

File: src/views/MapView.vue [hunk 2]
```diff
@@ -26,10 +28,37 @@ export default {
 };
 </script>
 <style lang="scss">
+$animOffset: -10rem;
```
Comment (blahmonkey) [inline src/views/MapView.vue]: @JatinThakur14 Put vars in a separate scss file and import it

**Review by blahmonkey — CHANGES_REQUESTED** [review]

File: src/components/MapComponent.vue (hunk 1 above)
Comment (cryogon) [reply src/components/MapComponent.vue:200]: Do you mean I shouldn't write path inside brackets?

File: src/components/LoginButton.vue [hunk 3]
```diff
@@ -31,7 +30,7 @@ export default {
 };
 </script>
 <style lang="scss">
-$btn-size: 3.3rem;
+@import "@/assets/style.scss";
```
Comment (blahmonkey) [inline src/components/LoginButton.vue]: @JatinThakur14 rename this to variables.scss - it isnt a style file

**Review by blahmonkey — CHANGES_REQUESTED** [review]

File: src/components/LoginButton.vue (hunk 3 above)
Comment (cryogon) [reply src/components/LoginButton.vue]: okay

**Review by blahmonkey — APPROVED** [review]

**Pull request merged**